great_expectations/experimental/datasources/data_asset/data_connector/file_path_data_connector.py

```diff
@@ -0,0 +1,293 @@
+from __future__ import annotations
+
+import logging
+import re
+from typing import TYPE_CHECKING, Callable, List, Optional
+
+from great_expectations.core.batch_spec import BatchSpec, PathBatchSpec
+from great_expectations.datasource.data_connector.batch_filter import (
+    BatchFilter,
+    build_batch_filter,
+)
+from great_expectations.datasource.data_connector.util import (
+    map_batch_definition_to_data_reference_string_using_regex,
+)
+from great_expectations.experimental.datasources.data_asset.data_connector import (
+    DataConnector,
+)
+from great_expectations.experimental.datasources.data_asset.data_connector.regex_parser import (
+    RegExParser,
+)
+
+# TODO: <Alex>ALEX_INCLUDE_SORTERS_FUNCTIONALITY_UNDER_PYDANTIC-MAKE_SURE_SORTER_CONFIGURATIONS_ARE_VALIDATED</Alex>
+# TODO: <Alex>ALEX</Alex>
+# from great_expectations.experimental.data_asset.data_connector.sorter import Sorter
+# TODO: <Alex>ALEX</Alex>
+from great_expectations.experimental.datasources.data_asset.data_connector.util import (
+    batch_definition_matches_batch_request,
+    map_data_reference_string_to_batch_definition_list_using_regex,
+)
+
+if TYPE_CHECKING:
+    from great_expectations.core.batch import BatchDefinition
+    from great_expectations.experimental.datasources.interfaces import BatchRequest
+
+
+logger = logging.getLogger(__name__)
+
+
+class FilePathDataConnector(DataConnector):
```

Comment:
Many of the interface methods can be removed

Reply:
@donaldheppner These interface methods are important for subclasses to implement; otherwise, we will not be able to guarantee performance and certain prescribed functionality.